### PR TITLE
Added finish button

### PR DESCRIPTION
**Pull request by mhib**
@adamzima @s1mplex @jtkpiotr @lkudyba

### CODE REVIEW

File: client/app/sources/EventSource.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ const EventSource = {
   },
   confirm(conferenceRoomId, eventId) {
     return axios.post(`${eventPath(conferenceRoomId)}/${eventId}/confirm`);
+  },
+  finish(conferenceRoomId, eventId) {
+    return axios.post(`${eventPath(conferenceRoomId)}/${eventId}/finish`);
```
Comment (jtkpiotr) [inline client/app/sources/EventSource.js]: How about:
const finishPath = (conferenceRoomId, eventId) => `${eventPath(conferenceRoomId)}/${eventId}/finish`

and then:
return axios.post(finishPath(conferenceRoomId, eventId));

File: client/test/components/conference_room/provider/EventProvider.spec.jsx [hunk 2]
```diff
@@ -106,4 +111,82 @@ describe('<EventProvider />', () => {
...
+
+    after(() => {
+      EventSource.confirm.restore();
+    });
+
+    context('with successful request', () => {
+      before(() => {
+        wrapper.setState({ currentEvent: Event.build({ confirmed: false }) });
+        EventSource.confirm.resolves();
+        componentWrapper.find('button#confirm').simulate('click');
```
Comment (jtkpiotr) [inline client/test/components/conference_room/provider/EventProvider.spec.jsx]: you can move `componentWrapper.find('button#confirm')` to a separate variable. This lets you stay DRY

File: client/test/components/conference_room/provider/EventProvider.spec.jsx [hunk 3]
```diff
@@ -106,4 +111,82 @@ describe('<EventProvider />', () => {
...
+      });
+
+      it('toggles current event\'s confirmation', () => {
+        expect(componentWrapper.prop('currentEvent').confirmed).to.be.true();
```
Comment (jtkpiotr) [inline client/test/components/conference_room/provider/EventProvider.spec.jsx:136]: `subject = componentWrapper.prop('currentEvent').confirmed`
and
`expect(subject).to.be.true()`

File: client/test/components/conference_room/provider/EventProvider.spec.jsx [hunk 4]
```diff
@@ -106,4 +111,82 @@ describe('<EventProvider />', () => {
...
+
+    after(() => {
+      EventSource.finish.restore();
+    });
+
+    context('with successful request', () => {
+      before(() => {
+        wrapper.setState({ currentEvent: Event.build() });
+        EventSource.finish.resolves();
+        componentWrapper.find('button#finish').simulate('click');
```
Comment (jtkpiotr) [inline client/test/components/conference_room/provider/EventProvider.spec.jsx]: as above

File: client/test/components/conference_room/provider/EventProvider.spec.jsx [hunk 5]
```diff
@@ -106,4 +111,82 @@ describe('<EventProvider />', () => {
...
+      });
+
+      it('removes current event', () => {
+        EventSource.finish.resolves();
+        componentWrapper.find('button#finish').simulate('click');
+        expect(componentWrapper.prop('currentEvent')).not.to.exist();
```
Comment (jtkpiotr) [inline client/test/components/conference_room/provider/EventProvider.spec.jsx:174]: extract this to subject or something

File: client/app/components/conference_room/provider/EventProvider.jsx [hunk 6]
```diff
@@ -77,6 +78,17 @@ export default class EventProvider extends React.Component {
...
   }
 
+  _finishEvent() {
+    if (typeof(this.state.currentEvent) === 'undefined') {
+      return;
+    }
+
+    const currentEvent = this.state.currentEvent;
+    this.setState({ currentEvent: undefined });
+    EventSource.finish(this.props.conferenceRoom.id, this.state.currentEvent.id)
```
Comment (jtkpiotr) [inline client/app/components/conference_room/provider/EventProvider.jsx]: I believe it should be `EventSource.finish(this.props.conferenceRoom.id, currentEvent.id)`

File: client/test/components/conference_room/provider/EventProvider.spec.jsx (hunk 3 above)
Comment (quirk0o) [reply client/test/components/conference_room/provider/EventProvider.spec.jsx:136]: I don't agree :P
`expect(subject).to.be.true()` doesn't say anything.

File: client/test/components/conference_room/provider/EventProvider.spec.jsx (hunk 5 above)
Comment (quirk0o) [reply client/test/components/conference_room/provider/EventProvider.spec.jsx:174]: No

File: client/app/components/conference_room/provider/EventProvider.jsx (hunk 6 above)
Comment (quirk0o) [reply client/app/components/conference_room/provider/EventProvider.jsx]: Done

File: client/app/sources/EventSource.js (hunk 1 above)
Comment (quirk0o) [reply client/app/sources/EventSource.js]: Done

File: client/test/components/conference_room/provider/EventProvider.spec.jsx (hunk 2 above)
Comment (quirk0o) [reply client/test/components/conference_room/provider/EventProvider.spec.jsx]: Done

File: client/test/components/conference_room/provider/EventProvider.spec.jsx (hunk 4 above)
Comment (quirk0o) [reply client/test/components/conference_room/provider/EventProvider.spec.jsx]: Done

**Review by jtkpiotr — APPROVED** [review]

**Pull request merged**